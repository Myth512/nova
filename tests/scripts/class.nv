class Vector {
	def _init_(x, y, z) {
		self.x = x
		self.y = y
		self.z = z
	}

	def _str_() {
		return "Vector({self.x}, {self.y}, {self.z})" 
	}

	def _add_(other) {
		return Vector(self.x + other.x, self.y + other.y, self.z + other.z)
	}

	def _sub_(other) {
		return Vector(self.x - other.x, self.y - other.y, self.z - other.z)
	}

	def _mul_(other) {
		return self.x * other.x + self.y * other.y + self.z + other.z
	}
}

v1 := Vector(1, 2, 3)

print(v1)

v1.w = 4

print(v1.w)

v2 := Vector(4, 5, 6)

print(v1 + v2)

print(v2 - v1)

print(v1 * v2)

