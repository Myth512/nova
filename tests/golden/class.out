Vector(1, 2, 3)
4
Vector(5, 7, 9)
Vector(3, 3, 3)
23
